### PR TITLE
DOC: Add into about empty results in sample

### LINKED ISSUE #28495: sample from Series should never return empty series
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd

# pandas 0.25
sample_data = pd.Series([1, 2])
result = sample_data.sample(frac=0.2)
result.empty

```
#### Problem description
When I sample with a fraction I always expect a result, returning an empty series is unexpected.

I found this issue in an older version of pandas, but it also happens in pandas 0.25

#### Expected Output
```python
Series([1], dtype: int64) # or 2 but not empty
```

#### Output of ``pd.show_versions()``

<details>
INSTALLED VERSIONS
------------------
commit           : None
python           : 3.6.8.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 18.7.0
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : en_US.UTF-8
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8
pandas           : 0.25.1
numpy            : 1.17.2
pytz             : 2019.2
dateutil         : 2.8.0
pip              : 18.1
setuptools       : 40.6.2
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
s3fs             : None
scipy            : None
sqlalchemy       : None
tables           : None
xarray           : None
xlrd             : None
xlwt             : None
xlsxwriter       : None
</details>

### PR DESCRIPTION
**Pull request by roelhogervorst**
As referred to in https://github.com/pandas-dev/pandas/issues/28495.

- [x] closes #28495
- [x] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by WillAyd** [comment]
Can you provide something more descriptive? This as is is rather ambiguous

File: pandas/core/generic.py [hunk 1]
```diff
@@ -4862,6 +4862,8 @@ def sample(
             Default = 1 if `frac` = None.
         frac : float, optional
             Fraction of axis items to return. Cannot be used with `n`.
+            Be careful, with a small number of rows and a small fraction the
+            result can be empty.
```
Comment (datapythonista) [inline pandas/core/generic.py]: Something like `If the computed number of samples is zero, the result will be empty.` would be better I think. It feels quite obvious to me, but if you find that surprising, happy to see it clarified.

**Review by WillAyd — CHANGES_REQUESTED** [review]
@roelhogervorst can you address comments?

**Comment by WillAyd** [comment]
Closing as stale but @roelhogervorst ping if you'd like to pick back up

**Pull request closed without merging**